src/awesome/keygrabber.rs

```
@@ -0,0 +1,78 @@
+//! AwesomeWM Keygrabber interface
+
+use ::lua::LUA;
+use rlua::{self, Lua, Table, Function, Value};
+use rustwlc::*;
+#[allow(deprecated)]
+use rustwlc::xkb::Keysym;
+
+pub const KEYGRABBER_TABLE: &str = "keygrabber";
+const SECRET_CALLBACK: &str = "__callback";
+
+/// Init the methods defined on this interface.
+pub fn init(lua: &Lua) -> rlua::Result<()> {
+    let keygrabber_table = lua.create_table();
+    keygrabber_table.set("run", lua.create_function(run))?;
+    keygrabber_table.set("stop", lua.create_function(stop))?;
+    keygrabber_table.set("isrunning", lua.create_function(isrunning))?;
+    let globals = lua.globals();
+    globals.set(KEYGRABBER_TABLE, keygrabber_table)
+}
+
+#[allow(deprecated)]
+/// Given the current input, handle calling the Lua defined callback if it is
+/// defined with the input.
+pub fn keygrabber_handle(mods: KeyboardModifiers, sym: Keysym, state: KeyState)
+                         -> rlua::Result<()> {
+    let lua = LUA.lock().expect("LUA was poisoned");
+    let lua_state = if state == KeyState::Pressed {
+        "press"
+    } else {
+        "release"
+    }.into();
+    let lua_sym = sym.get_name().expect("Key symbol did not have a defined name");
```

Should return an error here